### PR TITLE
migrate to kotlin

### PR DESCRIPTION
**Pull request by StevenWatremez**

### CODE REVIEW

**Review by anthonyGuiguen — APPROVED** [review]

File: android-library/build.gradle [hunk 1]
```diff
@@ -26,11 +25,13 @@ android {
...
+    implementation 'androidx.appcompat:appcompat:1.0.2'
     testImplementation 'junit:junit:4.12'
-    androidTestImplementation 'com.android.support.test:runner:1.0.1'
-    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
-    compile 'com.zetapush:android-client:2.7.3'
+    androidTestImplementation 'androidx.test:runner:1.1.1'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.1'
+    implementation 'com.zetapush:android-client:3.0.3'
+    implementation "androidx.core:core-ktx:1.0.1"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
```
Comment (jgoas) [inline android-library/build.gradle]: move this lib in jdk8 :)

File: android-library/src/main/java/com/zetapush/library/Client.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        get() = zetaPushService.resource
+        set(resource) {
+            zetaPushService.resource = resource
+        }
+
+    /**
+     * Constructor for a ZetaPush Client
+     */
+    constructor(context: Context) {
+        val storageTokenHandler = KeyValueTokenStorage(context, "key_storage_token")
```
Comment (jgoas) [inline android-library/src/main/java/com/zetapush/library/Client.kt:64]: Can you just apply one of these constructors below ?
like : 
```
constructor(context: Context) {
  val storageCredentialHandler = KeyValueCredentialsStorage(context, "key_storage_login", "key_storage_password")
  this(context, storageCredentialHandler)
}
```

File: android-library/src/main/java/com/zetapush/library/KeyValueCredentialsStorage.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    override val credentials: Map<String, String>
+        get() {
+            val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
+            val login = sharedPreferences.getString(keyLogin, null)
+            val password = sharedPreferences.getString(keyPassword, null)
+
+            val credentials = HashMap<String, String>()
+            login?.let { login ->
+                credentials["login"] = login
```
Comment (jgoas) [inline android-library/src/main/java/com/zetapush/library/KeyValueCredentialsStorage.kt]: Constant? And maybe you should return a Credentials Object?

Comment (jgoas) [reply android-library/src/main/java/com/zetapush/library/KeyValueCredentialsStorage.kt]: I prefer manipulate an object than a Hashmap

Comment (jgoas) [reply android-library/src/main/java/com/zetapush/library/KeyValueCredentialsStorage.kt]: but if you want more data in credential object, you can add a field, i.e: "data" with a Map type. :)

File: android-library/src/main/java/com/zetapush/library/StorageCredentialsInterface.kt [hunk 4]
```diff
@@ -1,28 +1,24 @@
...
- * Created by damien on 20/07/17.
- */
-
-public interface StorageCredentialsInterface {
+    /**
+     * Get the credentials as a map with key 'login' and 'password'
+     *
+     * @return
+     */
+    val credentials: Map<String, String>?
```
Comment (jgoas) [inline android-library/src/main/java/com/zetapush/library/StorageCredentialsInterface.kt]: I prefer Credential Object :)

File: android-library/src/main/java/com/zetapush/library/ZetaPushService.kt [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+
+    /**
+     * Listener for the ZetaPush Connection
+     */
+    private inner class ZPConnectionListener : ConnectionStatusListener {
+
+        override fun successfulHandshake(map: Map<String, Any>) {
+
+            Log.d("ZP", "successfulHandshake -> map = $map")
```
Comment (jgoas) [inline android-library/src/main/java/com/zetapush/library/ZetaPushService.kt:171]: Tag Contant?

File: android-library/src/main/java/com/zetapush/library/Client.kt (hunk 2 above)
Comment (StevenWatremez) [reply android-library/src/main/java/com/zetapush/library/Client.kt:64]: in the next PR you will see this improvement a lot more convenient than this

**Review by jgoas — APPROVED** [review]

**Review by jgoas — APPROVED** [review]

**Pull request merged**